cmd/kube-apiserver/app/server.go

```diff
@@ -106,7 +105,7 @@ const etcdRetryLimit = 60
 const etcdRetryInterval = 1 * time.Second
 
 // NewAPIServerCommand creates a *cobra.Command object with default parameters
-func NewAPIServerCommand() *cobra.Command {
+func NewAPIServerCommand(stopCh <-chan struct{}) *cobra.Command {
```

Comment:
it is not obvious to me why we are changing this in this pr.

it seems unrelated to the kubelet changes needing to pass a channel down to the dockershim.

Reply:
In `cmd/hyperkube/main` we setup the signal handler using `server.SetupSignalHandler()`. This function can only be called one; subsequent calls deliberately fail:

    func SetupSignalHandler() (stopCh <-chan struct{}) {
            close(onlyOneSignalHandler) // panics when called twice
            ...

`NewHyperKubeCommand()` directly calls `NewAPIServerCommand()` and prior to this PR `NewAPIServerCommand()` also called `SetupSignalHandler()`. Without removing the secondary registration in `NewAPIServerCommand()` the registration would happen twice, resulting in a `panic on closed channel`.

To fix this we now pass the stopCh in from the roots of the call trees rather than letting it happen part-way down in `NewAPIServerCommand()`. 

And this is why this PR also changed the following files:

```console
$ ag NewAPIServerCommand

File: cmd/genman/gen_kube_man.go
66:27:		apiserver := apiservapp.NewAPIServerCommand(server.SetupSignalHandler())

File: cmd/genkubedocs/gen_kube_docs.go
57:27:		apiserver := apiservapp.NewAPIServerCommand(server.SetupSignalHandler())

File: cmd/kube-apiserver/app/server.go
107:4:// NewAPIServerCommand creates a *cobra.Command object with default parameters
108:6:func NewAPIServerCommand(stopCh <-chan struct{}) *cobra.Command {

File: cmd/kube-apiserver/apiserver.go
41:17:	command := app.NewAPIServerCommand(server.SetupSignalHandler())
```